third_party/proton/dialect/include/Conversion/ProtonToProtonGPU/Passes.td

```diff
@@ -9,38 +9,66 @@ def ConvertProtonToProtonGPU: Pass<"convert-proton-to-protongpu", "mlir::ModuleO
+                    clEnumValN(gpu::Granularity::WARP, "warp", "Warp"),
+                    clEnumValN(gpu::Granularity::WARP_2, "warp-2", "2 Warps"),
+                    clEnumValN(gpu::Granularity::WARP_4, "warp-4", "4 Warps"),
+                    clEnumValN(gpu::Granularity::WARP_8, "warp-8", "8 Warps"),
+                    clEnumValN(gpu::Granularity::WARP_GROUP, "warp-group", "Warp Group"),
+                    clEnumValN(gpu::Granularity::WARP_GROUP_2, "warp-group-2", "2 Warp Groups"),
+                    clEnumValN(gpu::Granularity::WARP_GROUP_4, "warp-group-4", "4 Warp Groups"),
+                    clEnumValN(gpu::Granularity::WARP_GROUP_8, "warp-group-8", "8 Warp Groups")
+              )}]>,
+       Option<"samplingStrategy", "sampling-strategy",
+              "SamplingStrategy", /*default*/"SamplingStrategy::NONE",
+              "Profiling sampling strategy",
+              /*parser*/[{::llvm::cl::values(
+                    clEnumValN(SamplingStrategy::NONE, "none", "No Sampling"),
+                    clEnumValN(SamplingStrategy::SELECTIVE, "selective", "Selective Sampling")
+              )}]>,
+       Option<"samplingOptions", "sampling-options",
               "std::string", /*default*/"\"\"",
-              "comma-separated list of selected ids">,
-       Option<"maxSharedMem", "max-shared-mem",
-              "int32_t", /*default*/"0",
-              "max shared memory size per CTA">,
-       Option<"scratchMem", "scratch-mem",
-              "int32_t", /*default*/"32768",
-              "profiler global scratch memory size per CTA">,
-       Option<"alignment", "alignment",
-              "int32_t", /*default*/"128",
-              "profiler global scratch memory alignment">,
-       Option<"strategy", "strategy", "std::string", /*default*/"\"circular\"",
-              "profiler record strategy (circular or flush)">,
-       Option<"bufferType", "buffer-type", "std::string", /*default*/"\"shared_mem\"",
-              "internal buffer type (shared_mem or stack_mem) that backs the profiling data">,
+              "Profiling sampling options">,
+       Option<"bufferStrategy", "buffer-strategy", "gpu::BufferStrategy", /*default*/"gpu::BufferStrategy::CIRCULAR",
+              "Profiler buffer recording strategy (circular or flush)",
+              /*parser*/[{::llvm::cl::values(
+                    clEnumValN(gpu::BufferStrategy::CIRCULAR, "circular", "Circular Buffer"),
+                    clEnumValN(gpu::BufferStrategy::FLUSH, "flush", "Flush Buffer")
+              )}]>,
+       Option<"bufferType", "buffer-type", "gpu::BufferType", /*default*/"gpu::BufferType::SHARED",
+              "Internal buffer type (SHARED, GLOBAL, LOCAL) that stores the profiling data",
+              /*parser*/[{::llvm::cl::values(
+                    clEnumValN(gpu::BufferType::DEFAULT, "default", "Default Memory"),
+                    clEnumValN(gpu::BufferType::SHARED, "shared", "Shared Memory"),
```

Comment:
Do we need to be careful with the naming convention here. "Local" and "shared" are overloaded in LLVM.

Local is the AMD "shared" memory space, thread "private" is scratch/stack (https://llvm.org/docs/AMDGPUUsage.html#address-spaces)
Shared is the NV "shared" and local is scratch/stack (https://llvm.org/docs/NVPTXUsage.html#address-spaces)

Reply:
I suppose "local" is OK here if we want to stick with the NV terminology. As long as we are consistent and don't mix and match.